src/lib/operators/jit_operator/operators/jit_read_tuples.cpp

```diff
@@ -103,29 +212,101 @@ void JitReadTuples::before_chunk(const Table& in_table, const ChunkID chunk_id,
   }
 
+  const auto add_iterator = [&](auto it, auto type, const JitInputColumn& input_column, const bool is_nullalbe) {
+    using IteratorType = decltype(it);
+    using Type = decltype(type);
+    if (is_nullalbe) {
+      context.inputs.push_back(std::make_shared<JitReadTuples::JitSegmentReader<IteratorType, Type, true>>(
+          it, input_column.tuple_entry.tuple_index()));
+    } else {
+      context.inputs.push_back(std::make_shared<JitReadTuples::JitSegmentReader<IteratorType, Type, false>>(
+          it, input_column.tuple_entry.tuple_index()));
+    }
+  };
+
+  bool use_specialization = true;
+
+  std::vector<bool> segment_is_dictionary(_input_columns.size(), false);
+  for (const auto& value_id_expression : _value_id_expressions) {
+    // Check for each expression using value ids whether the corresponding segment is dictionary-encoded.
+    const auto& jit_input_column = _input_columns[value_id_expression.input_column_index];
+    const auto segment = in_chunk.get_segment(jit_input_column.column_id);
+    const auto dictionary = get_dictionary_segment(segment).dictionary_segment;
+    segment_is_dictionary[value_id_expression.input_column_index] = dictionary != nullptr;
+
+    if (dictionary) {
+      if (jit_expression_is_binary(value_id_expression.expression_type)) {
+        // Set the searched value id for each expression according to the segment's dictionary in the runtime tuple.
+
+        // Retrieve the searched value
+        AllTypeVariant value;
+        size_t tuple_index;
+        if (const auto literal_index = value_id_expression.input_literal_index) {
+          value = _input_literals[*literal_index].value;
+          tuple_index = _input_literals[*literal_index].tuple_entry.tuple_index();
+        } else {
+          const auto parameter_index = value_id_expression.input_parameter_index;
+          value = parameter_values[*parameter_index];
+          tuple_index = _input_parameters[*parameter_index].tuple_entry.tuple_index();
+        }
+
+        // Null values are set in before_query() function
+        if (variant_is_null(value)) continue;
+
+        // Convert the value to the column data type
+        const auto casted_value = convert_variant_to_data_type(value, jit_input_column.tuple_entry.data_type());
+
+        // Lookup the value id according to the comparison operator
+        ValueID value_id = get_search_value_id(value_id_expression.expression_type, dictionary, casted_value);
+        context.tuple.set<ValueID::base_type>(tuple_index, value_id);
+      }
+    } else {
+      use_specialization = false;
+    }
+  }
+
+  // If the specialized function cannot be used, the jit expressions must be updated according to the encoding of the
+  // current chunk
+  if (!use_specialization) {
+    for (const auto& value_id_expression : _value_id_expressions) {
+      if (segment_is_dictionary[value_id_expression.input_column_index]) {
+        _enable_use_of_value_ids_in_expression(value_id_expression);
```

Comment:
wouldn't set_... make the code below shorter?

Reply:
A little bit. I merged code of `_enable_use_of_value_ids_in_expression()` and `_disable_use_of_value_ids_in_expression()` into  `_set_use_of_value_ids_in_expression()` function.